pretrain/tinyllama.py

```diff
@@ -107,7 +106,14 @@ def main(fabric, resume):
     )
     optimizer = fabric.setup_optimizers(optimizer)
 
-    state = {"model": model, "optimizer": optimizer, "hparams": hparams, "iter_num": 0, "step_count": 0}
+    state = {
+        "model": model, 
+        "optimizer": optimizer, 
+        "train_dataloader": train_dataloader._dataloader,
```

Comment:
(not blocking) Does _FabricDataLoader state_dict not route it properly? We should fix that

Reply:
Oh it does indeed work. I thought it wouldn't because FabricDataLoader would not be recognized as an instance of `_Stateful` but, I forgot that we unwrap the `FabricDataLoader` in `fabric.save()` and `.load()`